admin_ui/backend/api/tools.py

```
@@ -242,6 +243,238 @@ async def preview_email_template(request: EmailTemplatePreviewRequest):
+                        timeout_ms=_safe_jsonable(item.get("timeout_ms")),
+                        output_variables=_safe_jsonable(item.get("output_variables")),
+                        parameters=params_out,
+                        has_input_schema=bool(item.get("has_input_schema", False)),
+                        source=source,
+                    )
+                )
+            return ToolCatalogResponse(tools=out, source="ai_engine")
+        except httpx.ConnectError:
+            continue
+        except Exception:
+            continue
```

Ruff (0.15.0)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)